src/SW_Output.c

```diff
@@ -2002,859 +2470,1232 @@ void SW_OUT_set_colnames(int tLayers, SW_VEGESTAB_INFO** parms,
+            continue;
+        }
+
+        for (i = 0; i < OutDom->used_OUTNPERIODS; i++) {
+            outPeriod = OutDom->timeSteps[k][i];
+            use_help = (Bool) (outPeriod != eSW_NoTime && writeit[outPeriod]);
+
+#ifdef STEPWAT
+            use_help_txt = use_help;
+            use_help_SXW = (Bool) (OutDom->timeSteps_SXW[k][i] != eSW_NoTime &&
+                                   writeit[OutDom->timeSteps_SXW[k][i]]);
+            use_help = (Bool) (use_help_txt || use_help_SXW);
+#endif
+
+            if (!use_help) {
+                continue; // don't call any `get_XXX` function
+            }
+
+#if defined(SOILWAT) && !defined(SWNETCDF)
+#ifdef SWDEBUG
+            if (debug) {
+                sw_printf(
+                    " call pfunc_text(%d=%s))", outPeriod, pd2str[outPeriod]
+                );
+            }
+#endif
+
+            ((void (*)(OutPeriod, SW_RUN *)
+            ) OutDom->pfunc_text[k])(outPeriod, sw);
+
+#elif defined(RSOILWAT) || defined(SWNETCDF)
+#ifdef SWDEBUG
+            if (debug) {
+                sw_printf(
+                    " call pfunc_mem(%d=%s))", outPeriod, pd2str[outPeriod]
+                );
+            }
+#endif
+            ((void (*)(OutPeriod, SW_RUN *, SW_OUT_DOM *)
+            ) OutDom->pfunc_mem[k])(outPeriod, sw, OutDom);
+
+#elif defined(STEPWAT)
+            if (use_help_SXW) {
+#ifdef SWDEBUG
+                if (debug) {
+                    sw_printf(
+                        " call pfunc_SXW(%d=%s))",
+                        OutDom->timeSteps_SXW[k][i],
+                        pd2str[OutDom->timeSteps_SXW[k][i]]
+                    );
+                }
+#endif
+                ((void (*)(OutPeriod, SW_RUN *, SW_OUT_DOM *)
+                ) OutDom->pfunc_SXW[k])(
+                    OutDom->timeSteps_SXW[k][i], sw, OutDom
+                );
+            }
+
+            if (!use_help_txt) {
+                continue; // SXW output complete; skip to next output period
+            } else {
+                if (OutDom->prepare_IterationSummary) {
+#ifdef SWDEBUG
+                    if (debug) {
+                        sw_printf(
+                            " call pfunc_agg(%d=%s))",
+                            outPeriod,
+                            pd2str[outPeriod]
+                        );
+                    }
+#endif
+                    ((void (*)(OutPeriod, SW_RUN *, SW_OUT_DOM *)
+                    ) OutDom->pfunc_agg[k])(outPeriod, sw, OutDom);
+                }
+
+                if (OutDom->print_SW_Output) {
+                    outPeriod = OutDom->timeSteps[k][i];
+#ifdef SWDEBUG
+                    if (debug) {
+                        sw_printf(
+                            " call pfunc_text(%d=%s))",
+                            outPeriod,
+                            pd2str[outPeriod]
+                        );
+                    }
+#endif
+                    ((void (*)(OutPeriod, SW_RUN *, SW_OUT_DOM *)

```

Comment:
STEPWAT2 code is calling the function pointed to by `pfunc_text` with an incorrect function type: it is `(void (*)(OutPeriod, SW_RUN *, SW_OUT_DOM *)` but should correctly be `(void (*)(OutPeriod, SW_RUN *)`